Added Solution

tlimato opened the pull request:
I made sure to explicitly state the sources I used to develop my solution. I will do this for all my other assignments I resubmit for review later today.

tlimato committed:
Added Solution

I made sure to explicitly state the sources I used to develop my solution. I will do this for all my other assignments I resubmit for review later today.

larskotthoff commented on README.md:
```diff
@@ -20,3 +21,74 @@ randomly without memory instead of systematically trying them?
+## Conclusion
+
+The time complexity of the algorithm is $\Theta(n!*n)$ because it generates all permutations of the input list and uses a for loop to iterate through each permutation and check whether or not it is sorted. The time complexity of the isSorted function is $\Theta(n)$ and the time complexity of generatePermutations is $\Theta(n!)$, so the overall time complexity of the algorithm is $\Theta(n!*n)$, g ...
+Simplifying for the most impactful part of the algorithm, the time complexity is $\Theta(n!)$.
```
You can't ignore non-constant factors. And what are the best/worst cases?

tlimato committed:
Update README.md

Removed The over simplification

tlimato committed:
revised solution

Added the requested best-case and worst-case analysis. I also removed my gross oversimplification of the runtime analysis.

tlimato commented:
I added the requested best-case and worst-case analysis. I also removed my gross oversimplification of the runtime analysis. I was tutoring at Step for Cosci 2 this past weekend where my tutees were just introduced to Big O notation and I just jumbled up the principles for a moment.
